deso/block.go

```diff
@@ -944,6 +956,161 @@ func (node *Node) GetValidatorPKIDFromSubAccountIdentifier(subAccount *types.Sub
+				continue
+			}
+			unstakeAmountNanosUint64 := unstakeAmountNanos.Uint64()
+			// First use an "input" from the ValidatorEntry
+			operations = append(operations, &types.Operation{
+				OperationIdentifier: &types.OperationIdentifier{
+					Index: int64(numOps),
+				},
+				Account: &types.AccountIdentifier{
+					Address:    lib.Base58CheckEncode(prevValidatorEntry.ValidatorPKID.ToBytes(), false, node.Params),
+					SubAccount: &types.SubAccountIdentifier{Address: ValidatorEntry},
+				},
+				Amount: &types.Amount{
+					Value:    strconv.FormatUint(unstakeAmountNanosUint64, 10),
+					Currency: &Currency,
+				},
+				Status: &SuccessStatus,
+				Type:   InputOpType,
+			})
+			numOps++
+			operations = append(operations, &types.Operation{
+				OperationIdentifier: &types.OperationIdentifier{
+					Index: int64(numOps),
+				},
+				Account: &types.AccountIdentifier{
+					Address: lib.Base58CheckEncode(prevStakeEntry.StakerPKID.ToBytes(), false, node.Params),
+					SubAccount: node.getLockedStakeEntrySubAccountIdentifierForValidator(
+						prevValidatorEntry.ValidatorPKID, 0), // TODO: we need some way to get the lockedAtEpochNumber. We'll likely need to expose it via utxo operations.
+				},
+				Amount: &types.Amount{
+					Value:    strconv.FormatUint(unstakeAmountNanosUint64, 10),
+					Currency: &Currency,
+				},
+				Status: &SuccessStatus,
+				Type:   OutputOpType,
+			})
+			numOps++
+		}
+	}
+	return operations
+}
+
+func (node *Node) getUnlockStakeOps(txn *lib.MsgDeSoTxn, utxoOps []*lib.UtxoOperation, numOps int) []*types.Operation {
+	if txn.TxnMeta.GetTxnType() != lib.TxnTypeUnlockStake {
+		return nil
+	}
+
+	var operations []*types.Operation
+	for _, utxoOp := range utxoOps {
+		if utxoOp.Type == lib.OperationTypeUnlockStake {
+			for _, prevLockedStakeEntry := range utxoOp.PrevLockedStakeEntries {
+				if !prevLockedStakeEntry.LockedAmountNanos.IsUint64() {
+					glog.Error("getUnlockStakeOps: lockedAmountNanos was not a uint64")
+					continue
+				}
+				lockedAmountNanosUint64 := prevLockedStakeEntry.LockedAmountNanos.Uint64()
+				// Each locked stake entry is an "input" to the UnlockStake txn
+				// We spend ALL the lockedAmountNanos from each locked stake entry in
+				// an unlocked transaction
+				operations = append(operations, &types.Operation{
+					OperationIdentifier: &types.OperationIdentifier{
+						Index: int64(numOps),
+					}, Account: &types.AccountIdentifier{
+						Address: lib.Base58CheckEncode(prevLockedStakeEntry.StakerPKID.ToBytes(), false, node.Params),
+						SubAccount: node.getLockedStakeEntrySubAccountIdentifierForValidator(
+							prevLockedStakeEntry.ValidatorPKID, prevLockedStakeEntry.LockedAtEpochNumber),
+					},
+					Amount: &types.Amount{
+						Value:    strconv.FormatUint(lockedAmountNanosUint64, 10),
+						Currency: &Currency,
+					},
+					Status: &SuccessStatus,
+					Type:   InputOpType,
+				})
+				numOps++
+			}
+		}
+	}
+	return operations
+}
```

Comment:
Aside from the lockedAtEpochNumber issue the above all seems like it should work.

Reply:
We will need to add stuff for the non-POS lockups but other than that it seems good.